### PR TITLE
Resize text on the map

### PR DESCRIPTION
**Pull request by matkoniecz**
This PR makes two things - text now gets larger as one zooms in, it is now also respecting device text size settings.

Both are inspired by #1872 

I admit that text now gets much larger on high zoom and I would not be surprised if it something disliked by @westnordost - in such case I may either retweaked to be subtler or entire "zoom in increases text" may be dropped (this mechanic is not a very noticeable but it actually appears in MAPS.ME, Google Maps, mapy.cz etc, so it is not something really weird).

--------

Sadly, I found no way to just pass scaling factor into map style (as say `global.text_size_scaling`) and do everything else there.

I tried things like that in map style itself and all of them failed - maybe I am missing something but it seems that it needs to be done outside map style file.

```
    text_size: [[15, global.text_size_scaling], [16, global.text_size_scaling * 2]]
    text_size: function() { return [[15, '12px'], [16, '50px']];}
    text_size: [[15, 12 * global.text_size_scaling], [16, 13 * global.text_size_scaling], [17, 15 * global.text_size_scaling], [18, 17 * global.text_size_scaling], [19, 24 * global.text_size_scaling]]
```

In case of this passing I will open PR against map style repo to add note about `text_size` and `language` getting overwritten by SC.

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/map/MapFragment.kt [hunk 1]
```diff
@@ -193,7 +193,13 @@ open class MapFragment : Fragment(),
     }
 
     protected open suspend fun getSceneUpdates(): List<SceneUpdate> {
-        return listOf(SceneUpdate("global.language", Locale.getDefault().language))
+        val textScale = getResources().getConfiguration().fontScale;
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/map/MapFragment.kt:196]: Don't you get suggestions/soft warnings here in Android Studio?

Comment (matkoniecz) [reply app/src/main/java/de/westnordost/streetcomplete/map/MapFragment.kt:196]: No, without any protests. Is there anything wrong/imperfect here?

**Comment by westnordost** [comment]
Not wrong, but usually Java getters are written like fields in Kotlin: 
resources.configuration.fontScale

Am 2. Juni 2020 03:25:20 MESZ schrieb Mateusz Konieczny <notifications@github.com>:
>@matkoniecz commented on this pull request.
>
>
>
>> @@ -193,7 +193,13 @@ open class MapFragment : Fragment(),
>     }
> 
>     protected open suspend fun getSceneUpdates(): List<SceneUpdate> {
>-        return listOf(SceneUpdate("global.language",
>Locale.getDefault().language))
>+        val textScale = getResources().getConfiguration().fontScale;
>
>No, without any protests. Is there anything wrong/imperfect here?

File: app/src/main/java/de/westnordost/streetcomplete/map/MapFragment.kt (hunk 1 above)
Comment (matkoniecz) [reply app/src/main/java/de/westnordost/streetcomplete/map/MapFragment.kt]: I reset Android Studio suggestion settings (I disabled some due to completely spurious reports) and it appeared as expected.

**Comment by matkoniecz** [comment]
https://github.com/ENT8R/streetcomplete-mapstyle/commit/5dcfd3226faffc68656daf26b4a0fe747bcb7cca#commitcomment-39604906 found a better solution - so it will be likely done by improving mapstyle (support for text scaling and scaling text) and once new version of mapstyle is merged then text scaling parameter will become available.

**Comment by westnordost** [comment]
OK, waiting for @ENT8R to push to streetcomplete-mapstyle

**Comment by westnordost** [comment]
So, only update of the map style is missing, right?

**Comment by matkoniecz** [comment]
Yes, once map style is pulled in it should work (I can retest it again once that happens to avoid things like `//` as yaml comments).

**Comment by westnordost** [comment]
Ok merged it to master

**Comment by matkoniecz** [comment]
Retested, works, as far as I can see it can be merged.

**Comment by westnordost** [comment]
Good, thanks!

**Pull request merged**